testing/internal/e2e/tests/base/auth_method_password_test.go

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+package base_test
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/hashicorp/boundary/api/authmethods"
+	"github.com/hashicorp/boundary/api/scopes"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCliAuthMethodPassword uses the boundary cli to create a new password auth
+// method, set it as the primary auth method for a scope, and log in to boundary
+// using a new account in that auth method.
+func TestCliAuthMethodPassword(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+
+	ctx := context.Background()
+	boundary.AuthenticateAdminCli(t, ctx)
+	newOrgId := boundary.CreateNewOrgCli(t, ctx)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary", e2e.WithArgs("scopes", "delete", "-id", newOrgId))
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+
+	// Create new auth method
+	output := e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"auth-methods", "create", "password",
+			"-scope-id", newOrgId,
+			"-name", "e2e Auth Method",
+			"-format", "json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	var newAuthMethodResult authmethods.AuthMethodCreateResult
+	err := json.Unmarshal(output.Stdout, &newAuthMethodResult)
+	require.NoError(t, err)
+	newAuthMethodId := newAuthMethodResult.Item.Id
+	// Need to manually clean up auth method until ICU-7833 is addressed
+	t.Cleanup(func() {
+		ctx := context.Background()
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary", e2e.WithArgs("auth-methods", "delete", "-id", newAuthMethodId))
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	t.Logf("Created Auth Method Id: %s", newAuthMethodId)
+
+	// Create account in auth method
+	testAccountName := "test-account"
+	newAccountId, acctPassword := boundary.CreateNewAccountCli(t, ctx, newAuthMethodId, testAccountName)
+
+	// Create user and attach new account
+	newUserId := boundary.CreateNewUserCli(t, ctx, newOrgId)
+	t.Cleanup(func() {
+		boundary.AuthenticateAdminCli(t, context.Background())
+		output := e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs("users", "delete", "-id", newUserId),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	boundary.SetAccountToUserCli(t, ctx, newUserId, newAccountId)
+
+	// Set new account as primary auth method for the new org
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"scopes", "update",
+			"-id", newOrgId,
+			"-primary-auth-method-id", newAuthMethodId,
+			"-format", "json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	var updateScopeResult scopes.ScopeUpdateResult
+	err = json.Unmarshal(output.Stdout, &updateScopeResult)
+	require.NoError(t, err)
+	require.Equal(t, newAuthMethodId, updateScopeResult.Item.PrimaryAuthMethodId)
```

Comment:
Apart from this scenario, do we want to also check that a new user is created based on a primary auth method when it's used for the first time?

doc: https://developer.hashicorp.com/boundary/docs/commands/scopes/update#primary-auth-method-id

Reply:
I'm not sure I follow what the expected behavior is there.

Reply:
Oh I see now...

Reply:
We could do this:
1. Create a login/password auth method and its account
2. Make the auth method primary
3. Login with that login/password
4. Login as admin
5. Check that a new user was created and the primary auth method is attached to it

Reply:
Added.